### ISSUE TITLE
Wrapping a trait with static methods fails

### ISSUE THREAD

**Issue by nwalfield** [althonos/blanket#12]
Consider:

```
use blanket;

#[blanket::blanket(derive(Ref))]
trait Foo {
    fn bar();

    fn bam(&self);
}

fn main() {
    println!("Hello, world!");
}
```

Compiling this results in:

```
$ cargo build
   Compiling blanket v0.1.0 (/tmp/blanket)
error: custom attribute panicked
 --> src/main.rs:3:1
  |
3 | #[blanket::blanket(derive(Ref))]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: called `Option::unwrap()` on a `None` value

error: could not compile `blanket` (bin "blanket") due to 1 previous error
```

It would be good if blanket could handle these traits.  At the very least, it would be nice if the error message were more informative.

**Comment by althonos** [comment]
Thanks for the report, I've started implementing it. I will probably add a check so that return types that contain `Self` are not allowed.

**Comment by althonos** [comment]
Should now be fixed in `v0.4.0`.
